### PR TITLE
Detect LongString headers

### LINKED ISSUE #109: Sometimes this library returns a ByteArrayLongString
We've received reports from the field that sometimes this lib returns a `com.rabbitmq.client.impl.LongStringHelper$ByteArrayLongString`. A lot of JRuby code doesn't know how to handle this since it doesn't extend `String` or any other common object.

I think the correct thing to do is to convert this to a RubyString. @michaelklishin what do you think?

### PR DESCRIPTION
**Pull request by andrewvc**
I think this Fixes #109 but I'm not sure how to test this. @michaelklishin are there any existing specs I could base one on that would make sense here?

### CODE REVIEW

**Comment by michaelklishin** [comment]
@andrewvc try putting a string longer than 255 characters into a header value.

**Comment by andrewvc** [comment]
@michaelklishin OK, I think this PR is ready to be reviewed / merged, I've got a good spec for it and fixed the root cause.

File: lib/march_hare/metadata.rb [hunk 1]
```diff
@@ -71,6 +74,25 @@ def redelivered?
     def redelivery?
       redeliver
     end
+    
+    # Turns LongString instances into String
+    private
+    LONG_STRING_TYPE=com.rabbitmq.client.LongString
```
Comment (michaelklishin) [inline lib/march_hare/metadata.rb]: Sorry to nitpick but can you please use spaces around the `=`?

File: lib/march_hare/metadata.rb [hunk 2]
```diff
@@ -71,6 +74,25 @@ def redelivered?
...
+    def deep_ls_fix(value)
```
Comment (michaelklishin) [inline lib/march_hare/metadata.rb]: I don't know how I feel about this name. How about `deep_normalize_headers`?

**Comment by andrewvc** [comment]
@michaelklishin just made those changes! BTW, should I bump the changelog and gemspec? Would we be able to make a new release for this? I think this is really bad for some LS users, so it'd be appreciated if possible!

**Comment by michaelklishin** [comment]
No need to change the version.

**Comment by michaelklishin** [comment]
@andrewvc there was a couple of lines where the name wasn't updated and failing specs reflected that. I corrected them and released [2.22.0](https://rubygems.org/gems/march_hare/versions/2.22.0-java). Thank you!

**Comment by andrewvc** [comment]
Thanks so much @michaelklishin !

**Pull request merged**